React/lifecycle-methods.md

```diff
@@ -0,0 +1,82 @@
+# Lifecycle Methods
+
+## Learning Goals
+- Learn about the three phases: mount, update and unmount
```

Comment:
I like this lesson. Looks like this will be followed by talking about APIs the same day, right? I think that's important for giving them a reason to care.